index.html

```diff
@@ -1615,6 +1618,211 @@ <h3>
+            </li>
+            <li>If validation failed, [=iteration/continue=].
+            </li>
+            <li>
+              Set |entry|["scope"] to |resolved_scope|.
+            </li>
+            <li>[=list/Append=] |entry| to |processedScopeExtensions|.
+            </li>
+          </ol>
+        </li>
+      </ol>
+      <aside class="note">
+        `origin` is the only type of scope extension entry that is currently
+        specified. Other types may be added in the future to support different
+        use cases. 
+      </aside>
+      </section>
+      <section class="informative">
+        <h3>The web-app-origin-association file</h3>
+        <p>A <dfn>web-app-origin-association</dfn> file is a JSON file that can
+          be used to [=validate scope extensions=]. It confirm an association
+          between the origin it is in with one or more web applications. It 
+          identifies apps uniquely by referencing manifest [=manifest/ids=]. 
+        </p>
+        <p>
+          Given [=origin=] |origin:origin|, a [=web-app-origin-association=]
+          file is expected to be downloadable from 
+          <code>[origin]/.well-known/web-app-origin-association</code>.
+        </p>
+      </section>
+      <section>
+      <h2>Validating scope extensions</h2>
+      <p>
+        To <dfn>validate scope extensions</dfn>, given [=ordered map=] 
+        |extension:ordered map|, [=URL=] |manifest_id:URL| and [=URL=] 
+        |resolved_scope:URL|:
+      </p>
+      <ol class="algorithm">
+        <li>Set |resolved_scope| to empty URL.</li>
+        <li>Let [=URL=] |download_url| be the result of parsing 
+          "/.well-known/web-app-origin-association" using |extension|["origin"] 
+          as the base [=URL=].
+        </li>
+        <li>Let [=ordered map=] |json:ordered map| be the result of downloading
+          |download_url|. 
+        </li>
+        <li>If the download fails, return false.
+        </li>
+        <li>If |json|[|manifest_id|] does not exist or is not an 
+          [=ordered map=], return false.</li>
+        <li>Let [=string=] |scope_path:string| be "/".</li>
+        <li>If |json|[|manifest_id|]["scope"] exists and is a [=string=], set
+          |scope_path| to |json|[|manifest_id|]["scope"]. 
+        </li>
+        <li>Set |resolved_scope:URL| be the result of parsing |scope_path| with 
+          |extension|["origin"] as the base URL.
+        </li>
+        <li>return true.</li>
+      </ol>
+    </section>
+    <section>
+      <h2>Applying scope extensions</h2>
+      <p>The user agent MAY choose to apply <dfn>apply extended scope</dfn> in
+        some scenarios and [=manifest/scope=] in others. 
+      </p>
+      <p>
+        If the [=application context=]'s [=navigable/active document=]'s [=URL=]
+        is not [=manifest/within scope=] but is [=within extended scope=], the
+        user agent SHOULD provide UI that allows the user to determine the
```

Comment:
this is a contractual thing. I don't think Chromium implementation would be in-spec. What do you mean by the 'determine the url or at least it's origin" here.

Where did this requirement come from?